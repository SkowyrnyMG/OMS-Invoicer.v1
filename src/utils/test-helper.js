import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
