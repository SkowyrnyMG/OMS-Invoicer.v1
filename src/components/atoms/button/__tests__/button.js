import React from 'react';
import { snapShotTest } from 'utils/tests/test-helper';

import Button from '../button';

describe('Button', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<Button />);
  });
});
