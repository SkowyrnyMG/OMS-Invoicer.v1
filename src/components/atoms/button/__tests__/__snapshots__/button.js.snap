// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should display in the document with correct styles 1`] = `
.c0 {
  padding: 0.8rem 1rem;
  width: 16rem;
  font-size: inherit;
  text-align: left;
  color: #FFFFFF;
  background-color: #446DF6;
  border: none;
  outline: none;
  border-radius: 0.5rem;
  cursor: pointer;
  -webkit-transition: 0.25s -webkit-transform;
  -webkit-transition: 0.25s transform;
  transition: 0.25s transform;
}

.c0:hover,
.c0:active {
  -webkit-transform: translateX(2px);
  -ms-transform: translateX(2px);
  transform: translateX(2px);
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0::first-letter {
  text-transform: uppercase;
}

@media (max-width:767px) {
  .c0 {
    width: 100%;
  }
}

<button
  class="c0"
/>
`;
